lib/node_modules/@stdlib/array/base/reshape/README.md

```diff
@@ -0,0 +1,117 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2025 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# reshape
+
+> Reshape an n-dimensional nested array into another m-dimensional having a desired shape.
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var reshape = require( '@stdlib/array/base/reshape' );
+```
+
+#### reshape( x, fromShape, toShape, colexicographic )
+
+Reshapes an n-dimensional nested array into another m-dimensional having a desired shape.
+
+```javascript
+var x = [ [ 1, 2, 3 ], [ 4, 5, 6 ] ];
+
+var out = reshape( x, [ 2, 3 ], [ 3, 2 ], false );
+// returns [ [ 1, 2 ], [ 3, 4 ], [ 5, 6 ] ]
+```
+
+To reshape in colexicographic order, provide a third argument equal to `true`.
+
+```javascript
+var x = [ [ 1, 2, 3 ], [ 4, 5, 6 ] ];
+
+var out = reshape( x, [ 2, 3 ], [ 3, 2 ], true );
+// [ [ 1, 4 ], [ 2, 5 ], [ 3, 6 ] ]
+```
+
+</section>
+
+<!-- /.usage -->
+
+<section class="notes">
+
+## Notes
+
+-   The function assumes that the `fromShape` and the `toShape` have the same number of dimensions.
```

Comment:
Why is this restriction necessary? Rather than the same number of dimensions, what we need to ensure is that the two shapes have the same number of elements.